File: app/ColorList.py
from app.BaseList import BaseList
from app.Color import Color
import csv

class ColorList(BaseList):
    def read_from_csv(self, filename):
        with open(filename, mode='r', newline='', encoding='utf-8') as file:
            reader = csv.reader(file)
            for row in reader:
                if len(row) >= 2:
                    color = Color(int(row[0]), row[1])
                    self.add(color)

    def get_as_xml(self):
        xml_items = [item.get_as_xml() for item in self._items]
        return "<colors>\n" + "\n".join(xml_items) + "\n</colors>"

File: ruby_app/models/color_list.rb
require_relative 'base_list'
require_relative 'color'
require_relative '../config/database'

class ColorList < BaseList
  def read_from_db
    @items = DB[:colors].order(:id).map do |r|
      Color.new(r[:id], r[:name])
    end
  end
end
